### PR TITLE
DNM: helm integration test

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
testing a happy path before refining

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
I have read the CLA Document and I hereby sign the CLA

File: tests/testutils/telemetry/common.go [hunk 1]
```diff
@@ -76,7 +76,7 @@ func populateDirectives(attrs *map[string]any) {
 			pattern := subs[1]
 			rec, err := regexp.Compile(pattern)
 			if err != nil {
-				panic(fmt.Errorf("failed compiling resource attributes RE2: %w", err))
+				panic(fmt.Errorf("failed compiling resource attributes RE2 (%s): %w", k, err))
```
Comment (atoulme) [inline tests/testutils/telemetry/common.go:79]: can this improvement be split in its own PR?

File: tests/testutils/kubeutils/kind_cluster.go [hunk 2]
```diff
@@ -96,9 +101,16 @@ func (k *KindCluster) Create() {
...
+	for _, pod := range []string{
+		"kube-apiserver", "kube-controller-manager", "kube-scheduler", "etcd-cluster", "coredns", "kindnet", "kube-proxy",
+	} {
+		k.WaitForPods(fmt.Sprintf("%s-.*", pod), "kube-system", 2*time.Minute)
+	}
+	k.WaitForPods("local-path-provisioner-.*", "local-path-storage", 2*time.Minute)
 }
 
-func (k *KindCluster) Delete() {
+func (k *KindCluster) Teardown() {
```
Comment (atoulme) [inline tests/testutils/kubeutils/kind_cluster.go:113]: This change is fine, but can it be split to its own PR so it can reduce the diff of this PR?
